src/index.spec.ts

```diff
@@ -0,0 +1,168 @@
+import 'mocha';
+import { assert } from 'chai';
+import { App, ExpressReceiver } from './index';
+import serverless from 'serverless-http';
+import {
+  delay,
+  createFakeMessageEvent,
+  createEventRequest,
+  importAppWithMockSlackClient,
+} from "./test-helpers";
+
+// How I'd like to write apps using koa-style middleware
+// @ts-ignore - This and the rest are changing signatures for consistency w/ koa, this wont run
+function exampleToContainScope() {  // tslint:disable:typedef
+  const app = new App();
+
+  // @ts-ignore
+  app.use(async (ctx, next) => {
+    const start = Date.now();
+    await next();
+    const end = Date.now();
+
+    console.log(`Request took ${end - start}ms`);
+  });
+
+  // @ts-ignore
+  app.message(
+      'Oh Hai!',
+      // @ts-ignore
+      async (ctx, next) => {
+        const start = Date.now();
+        await next();
+        const end = Date.now();
+
+        console.log(`Message request took ${end - start}ms`);
+      },
+      // @ts-ignore
+      async ({ say, body }, next) => {
+        await say(`You said ${body.text}`);
+        await next();
+      },
+    );
+
+  app.message('Throw an error', async () => {
+    throw new Error('Wow, such error!');
+  });
+
+  const doAsyncProcessing = async () => {
+    await delay(100);
+
+    return 'Wow that took a while to do that thing!';
+  };
+
+  // @ts-ignore - changing the signature for consistency w/ koa
+  app.action(async ({ ack, respond }, next) => {
+    const [asyncProcessingResult] = await Promise.all([doAsyncProcessing(), ack()]);
+
+    await respond(asyncProcessingResult);
+    await next();
+  });
+
+  app.error(async (error) => {
+    console.error(error);
+    // Possibly attempt some graceful async shutdowns
+  });
```

Comment:
couldn't you write this currently as
```suggestion
  app.error(console.error));
```
?

Reply:
Yes, but it doesn't show the point of doing async work here which was the intent of the comment. I'll add code.